linkerd/http/retry/src/lib.rs

```diff
@@ -300,8 +305,9 @@ async fn send_req_with_retries(
             tracing::debug!("Retry success");
             return result.map(|rsp| rsp.map(BoxBody::new));
         }
-        if backup.body().is_capped() {
-            return result.map(|rsp| rsp.map(BoxBody::new));
+        match backup.body().is_capped() {
+            Some(false) => {}
+            None | Some(true) => return result.map(|rsp| rsp.map(BoxBody::new)),
```

Comment:
golf/tioli:

```suggestion
        if matches!(backup.body().is_capped(), None | Some(true)) {
            return result.map(|rsp| rsp.map(BoxBody::new));
```

Reply:
426017d applies this transformation to both of the relevant `match` blocks.